domain/miningmanager/mempool/revalidate_high_priority_transactions.go

```diff
@@ -6,22 +6,35 @@ import (
 	"github.com/kaspanet/kaspad/infrastructure/logger"
 )
 
-func (mp *mempool) revalidateHighPriorityTransactions() ([]*externalapi.DomainTransaction, error) {
+func (mp *mempool) revalidateHighPriorityTransactions(clone bool) ([]*externalapi.DomainTransaction, error) {
 	onEnd := logger.LogAndMeasureExecutionTime(log, "revalidateHighPriorityTransactions")
 	defer onEnd()
 
 	validTransactions := []*externalapi.DomainTransaction{}
+	if !clone {
+		for _, transaction := range mp.transactionsPool.highPriorityTransactions {
+			isValid, err := mp.revalidateTransaction(transaction)
+			if err != nil {
+				return nil, err
+			}
+			if !isValid {
+				continue
+			}
 
-	for _, transaction := range mp.transactionsPool.highPriorityTransactions {
-		isValid, err := mp.revalidateTransaction(transaction)
-		if err != nil {
-			return nil, err
-		}
-		if !isValid {
-			continue
+			validTransactions = append(validTransactions, transaction.Transaction())
```

Comment:
I think it's better to move the `clone` if-else here

Reply:
i will make it clone by default as per other comment.